app/api/crud.py

```diff
@@ -148,20 +156,36 @@ async def get_terms(data_element_URI: str):
     dict
         Dictionary where the key is the Neurobagel class and values correspond to all the unique terms representing available (i.e. used) instances of that class.
     """
-    cross_node_results = []
-    params = {data_element_URI: data_element_URI}
+    node_errors = []
+    unique_terms_dict = {}
 
-    for node_url in util.FEDERATION_NODES:
-        response = util.send_get_request(
+    params = {data_element_URI: data_element_URI}
+    tasks = [
+        util.send_get_request(
             node_url + "attributes/" + data_element_URI, params
         )
+        for node_url in util.FEDERATION_NODES
+    ]
+    responses = await asyncio.gather(*tasks, return_exceptions=True)
 
-        cross_node_results.append(response)
-
-    unique_terms_dict = {}
+    for node_url, response in zip(util.FEDERATION_NODES, responses):
+        node_name = util.FEDERATION_NODES[node_url]
+        if isinstance(response, HTTPException):
+            node_errors.append(
+                {"node_name": node_name, "error": response.detail}
+            )
+            warnings.warn(
+                f"Request to node {node_name} ({node_url}) did not succeed: {response.detail}"
+            )
+        else:
+            # Build the dictionary of unique term-label pairings from all nodes
+            for term_dict in response[data_element_URI]:
+                unique_terms_dict[term_dict["TermURL"]] = term_dict
```

Comment:
If we overwrite previous instances here, would it be reasonable to check first if the term is the same as the one we've already stored? If we ever encounter a situation where some jokester has a neurobagel node where they just make up their own TermURLs for our terms, we could end up with a tricky to debug situation if all that determines what we return here is the order in which we unpack the responses from nodes. Wdyt?

Reply:
I just saw that this is old code, so let's not address this here. I'll make an issue to discuss in general

Reply:
Thanks for catching this. I couldn't tell if you already created this issue so here's a new one! https://github.com/neurobagel/federation-api/issues/68

Reply:
Indeed, I had not! Thanks for catching that :)